examples/hello-world/hello-pt/jobs/hello-pt/app/custom/cifar10validator.py

```diff
@@ -48,6 +53,11 @@ def __init__(self, data_path="~/data", validate_task_name=AppConstants.TASK_VALI
         test_data = CIFAR10(root=data_path, train=False, transform=transforms)
         self._test_loader = DataLoader(test_data, batch_size=4, shuffle=False)
 
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == EventType.START_RUN:
+            data_path = os.path.join(self.data_path, fl_ctx.get_identity_name())
```

Comment:
There is good and bad about adding this additional "fl_ctx.get_identity_name()" to it.
Since this data_path is user's passed in, IMO would be better to use it directly instead of adding a sub-folder to it.
Another thing, for the new code, if the data_path is not writable we can not run this code.

On the other hand you could argue that by adding this sub-folder, we remove the burden to ask user to run prepare_data.sh in the hello example.

Reply:
fl_ctx.get_identity_name() is to use the unique client_name. This is to avoid the concurrent clients overwrite the data in the same place.